tools/auto/presenceValidator.ts

```diff
@@ -0,0 +1,280 @@
+import "source-map-support/register.js";
+
+import actions from "@actions/core";
+import { existsSync, readFileSync } from "fs";
+import { readFile, stat } from "fs/promises";
+import got from "got";
+import jsonAst, { ObjectNode } from "json-to-ast";
+import { validate } from "jsonschema";
+import { resolve } from "path";
+import { compare } from "semver";
+
+import PresenceCompiler from "../classes/PresenceCompiler.js";
+import getDiff from "../util/getDiff.js";
+import getLatestSchema from "../util/getLatestSchema.js";
+
+const schema = await getLatestSchema(),
+	changedPresences = getDiff(),
+	validLangs = await getValidLanguages(),
+	compiler = new PresenceCompiler(),
+	errors: {
+		presence: string;
+		message: string | Error;
+		properties?: actions.AnnotationProperties | undefined;
+	}[] = [];
+
+for (const presence of changedPresences) {
+	const presencePath = compiler.getPresenceFolder(presence);
+
+	//#region Metadata Check
+	if (!(await stat(resolve(presencePath, "metadata.json"))).isFile()) {
+		errors.push({
+			presence,
+			message: "Presence is missing metadata.json!",
```

Comment:
couldn't find metadata in ${presence}

Reply:
Error has a file linked, meaning the summary shows the file path.

Reply:
Remove the `!